### DISCUSSION TITLE
expose restrict editing/git lfs checkboxes in the api?

### DISCUSSION THREAD

**Discussion by ljharb** [community/community#126460] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

I'm trying to build some tooling to manage multiple repos (across users/orgs), and it seems like [this checkbox](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request) isn't present in the api results. Can it be added?

It also seems like the "Include Git LFS objects in archives" checkbox and the wiki "Restrict editing to collaborators" checkbox are missing from the api as well.

(in general, it'd be really nice if nothing was allowed to be added to the UI without also adding it to the API, even if that didn't include allowing it to be changed via the API)

If this isn't the right place to file this, please point me there, and I'll be happy to move it :-)

(continuation from https://github.com/community/maintainers/discussions/260)

**Comment by zpao** [comment]
FWIW this seems to be there in the [GraphQL API](https://docs.github.com/en/graphql/reference/mutations#enablepullrequestautomerge) for mutation (and reading - `autoMergeRequest` on the `PullRequest` object). It seems like it may be there for reading in the REST API as well (the undocumented `allow_auto_merge` field) but not modifying.

> (in general, it'd be really nice if nothing was allowed to be added to the UI without also adding it to the API, even if that didn't include allowing it to be changed via the API)

Seconded. I'll keep advocating for that from our business relationship too.

The lack of consistency between GraphQL and REST APIs is also a huge problem. Switching between these for specific uses is really annoying. I've been pressing on that one as well.

**Reply by ljharb** [reply]
oh awesome, thanks - i use the graphql API but i agree they should be consistent.

**Reply by ljharb** [reply]
actually - if it's on the PR object, does that reflect the button on the PR itself? This issue's asking about a field on the Repository object.

**Reply by zpao** [reply]
Ah misread. It looks like `autoMergeAllowed` is on the `Repository` object. I don't see it as a field on the `updateRepository` mutation though and the only related mutation otherwise that I saw was for the PR itself :(

**Reply by ljharb** [reply]
looks like AutoMergeAllowed indeed exists already, but i've retitled this because of the other two.
